internal/oci/container.go

```diff
@@ -267,14 +275,19 @@ func (c *Container) Metadata() *pb.ContainerMetadata {
 
 // State returns the state of the running container
 func (c *Container) State() *ContainerState {
-	c.opLock.RLock()
-	defer c.opLock.RUnlock()
-	return c.state
+	c.stateLock.RLock()
+	defer c.stateLock.RUnlock()
+	if c.state != nil {
+		cs := *c.state
+		return &cs
+	}
+	return nil
 }
 
-// StateNoLock returns the state of a container without using a lock.
-func (c *Container) StateNoLock() *ContainerState {
-	return c.state
+func (c *Container) Pid() int {
+	c.stateLock.RLock()
```

Comment:
The pid also shouldn't change after we set it for the container. We though need to make sure around the runc state calls that return stopped status.

Reply:
@mrunalp 
The read lock is taken out.